flamedisx/likelihood.py

```diff
@@ -267,6 +274,7 @@ def set_data(self,
     def simulate(self, fix_truth=None, **params):
         """Simulate events from sources.
         """
+        params = self.prepare_params(params, free_all_rates=True)
```

Comment:
Is this to clear the free rates param from the guess when simulating so you don't need to pop the free rate args from best fit when given to the simulation?

Reply:
By default, only the rate multipliers of the sources in `free_rates` are legal parameters to the likelihood, and only those will appear in bestfit results. It can be useful for users of `.simulate` to be able to vary all rates, even those of fixed sources. If nothing else, we use this in the unit tests (https://github.com/FlamTeam/flamedisx/blob/master/tests/test_likelihood.py#L205). Not sure if this answers your question -- maybe you're thinking of a particular case where you are passing best-fit results to a simulation?

Reply:
Ok, yes, I was mistakenly thinking of the bf simulation. I think I get it now. Thanks!